src/entities/road.lua

```diff
@@ -86,10 +86,39 @@ function road:draw()
   love.graphics.setColor(255, 255, 255, 255)
 end
 
-function road:update(angle, roadShift)
+  
+function road:update(newSegments)
   -- create a road segment at the position and angle of the frontier.
-  self:addsegment()
+  for _, s in pairs(newSegments) do
```

Comment:
use ipairs here to guarantee that we iterate in order.  pairs can iterate the keys in any order.